File: driver-service/src/main/java/com/example/driverservice/service/CarParkService.java
package com.example.driverservice.service;

import com.example.driverservice.dto.request.CarOwnerRequest;
import com.example.driverservice.dto.response.CarResponse;

public interface CarParkService {

    CarResponse getCarById(Long id);

    CarResponse setOwner(CarOwnerRequest carOwnerRequest);

}

File: car-park-service/src/main/kotlin/org/example/carparkservice/service/CarParkService.kt
package org.example.carparkservice.service

import org.example.carparkservice.dto.CarListResponse
import org.example.carparkservice.dto.CarOwnerRequest
import org.example.carparkservice.dto.CarRequest
import org.example.carparkservice.dto.CarResponse
import org.springframework.security.oauth2.jwt.Jwt


interface CarParkService {
    fun getCarById(id: Long): CarResponse
    fun createCar(car: CarRequest): CarResponse?
    fun deleteCarById(id: Long): CarResponse?
    fun findAllCars(): CarListResponse?
    fun setDriverToCar(carOwnerRequest: CarOwnerRequest): CarResponse?
    fun extractUserInfo(jwt: Jwt): Any?


}